cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java

```diff
@@ -888,35 +882,70 @@ public KafkaBrokerConfigurationBuilder withUserConfiguration(KafkaConfiguration
         return this;
     }
 
-    /**
-     * Adds the Kafka metric.reporters to the user configuration.
-     *
-     * @param userConfig The user configuration to which the metric reporters will be added.
-     * @param injectCcMetricsReporter Flag indicating whether to inject the Cruise Control Metrics Reporter.
-     * @param injectKafkaJmxReporter  Inject the JMX Reporter into the configuration
-     * @param injectStrimziMetricsReporter Flag indicating whether to inject the Strimzi Metrics Reporter.
-     */
-    private void maybeAddMetricReporters(KafkaConfiguration userConfig, boolean injectCcMetricsReporter, boolean injectKafkaJmxReporter, boolean injectStrimziMetricsReporter) {
-        if (injectCcMetricsReporter) {
-            createOrAddListConfig(userConfig, "metric.reporters", CruiseControlMetricsReporter.CRUISE_CONTROL_METRIC_REPORTER);
-        }
-        if (injectKafkaJmxReporter) {
-            createOrAddListConfig(userConfig, "metric.reporters", "org.apache.kafka.common.metrics.JmxReporter");
-        }
-        if (injectStrimziMetricsReporter) {
-            createOrAddListConfig(userConfig, "metric.reporters", "io.strimzi.kafka.metrics.KafkaPrometheusMetricsReporter");
+    private void printMetricReporters(KafkaConfiguration userConfig,
+                                      boolean injectCcMetricsReporter,
+                                      boolean injectKafkaJmxReporter,
+                                      boolean injectStrimziMetricsReporter) {
+        // Build a list of reporters to inject based on flags
+        List<String> reportersToInject = new ArrayList<>();
+        if (injectCcMetricsReporter) reportersToInject.add(CruiseControlMetricsReporter.CRUISE_CONTROL_METRIC_REPORTER);
+        // Since Kafka 4 the JmxReporter is explicitly added when JmxPrometheusExporter is enabled
+        if (injectKafkaJmxReporter) reportersToInject.add("org.apache.kafka.common.metrics.JmxReporter");
+        if (injectStrimziMetricsReporter) reportersToInject.add(StrimziMetricsReporterConfig.KAFKA_CLASS);
+
+        if (!reportersToInject.isEmpty()) {
+            String configKey = "metric.reporters";
+
+            if (userConfig != null && !userConfig.getConfiguration().isEmpty() &&
+                    userConfig.getConfigOption(configKey) != null) {
+                // handle user configuration if present and avoids duplicates
+                String configValue = userConfig.getConfigOption(configKey);
+
+                reportersToInject = reportersToInject.stream()
+                        .filter(r -> !userConfig.getConfigOption(configKey).contains(r))
+                        .toList();
+
+                if (!reportersToInject.isEmpty()) {
+                    userConfig.removeConfigOption(configKey);
+
+                    printSectionHeader(configKey + " configuration");
```

Comment:
Isn't it better "Metrics Reporters configuration" here?

Reply:
Metrics Reporters is a bit overloaded in this context, so using the exact configuration name is better IMO.

Reply:
It's going to be the only section header with a property name while the section title should be something to group stuff and more readable ... at this point we can even remove it if we are not able to find a good section title.

Reply:
I proposed "Metrics Reporters configuration". Why do you think it's overloaded here?

Reply:
in this end for the last two this is what `metric.reporters` and `kafka.metric.reporters` fields are, or?
I agree that now we have a "Strimzi" one but ... :-D

> That said, if you prefer like this, it is fine with me.

I am not saying I won't approve if you don't change. I don't want to start another long discussion with other maintainers. So it's up to you ... I am going to approve anyway. It's a minimal thing.

Reply:
The problem here is in Kafka, where these configuration names are ambiguous. I just think that there is less space for confusion using verbatim configuration names.